specification/sbi_atee.adoc

```diff
@@ -101,35 +101,107 @@ This operation requires the host to convert a sufficient number of non-confident
+     * and VS CSRs.
+     *
+     * These CSRs may be updated by `sbi_tee_host_run_tvm_vcpu()` in the TEE-Host extension. See
+     * the documentation of `sbi_tee_host_run_tvm_vcpu()` for more details.
+     */
+    uint64_t csrs[1024];
+};
+
+struct tsm_shmem_scratch {
+    /* General purpose registers for a TVM guest.
+     *
+     * The TSM will always read or write the minimum number of registers in this set to complete
+     * the requested action. To avoid leaking information from the TVM, the TSM must follow the
+     * given rules.
+     *
+     * The TSM will write to these registers upon return from `sbi_tee_host_run_tvm_vcpu()` when:
+     *  - The vCPU takes a store guest page fault in an emulated MMIO region.
+     *  - The vCPU makes an ECALL that is to be forwarded to the host.
+     *
+     * The TSM will read from these registers when:
+     *  - The vCPU takes a load guest page fault in an emulated MMIO region.
+     */
+    uint64_t guest_gprs[32];
+    uint64_t _reserved[224];
+};
+-------------------
+
+Given is the list of CSRs and GPRs that the TSM and the host are supposed to use from NACL shared memory. Also given
+are the allowed operation for each software in terms of `R` (read) and `W` (write). For example, the host is only
+supposed to read `htinst`. The host can write to it without any problems but the TSM is not supposed to read that
+value as it's only supposed to write to it and ignore any modifications.
+
+[#table_tsm_csr_updates_in_nacl]
+.TSM NACL CSRs and GPRs
+[cols="2,1,1,12", width=100%, align="center", options="header"]
+|===
+| *CSRs*     | *TSM* | *Host* | *Purpose*
+| htinst     |  W    |  R     | In the case of MMIO emulation, TSM writes a modified form of load/store
+                                instruction in htinst CSR to allow the host to emulate the MMIO.
+| htval      |  W    |  R     | In case of a guest page-fault, TSM writes the guest's physical address into
+                                htval CSR.
+| htimedelta |  W    |  R     | TSM writes the guest htimedelta in this CSR. This is to allow the host to schedule an internal
+                                software timer for the guest to keep the timer interrupt ticking.
+| vstimecmp  |  W    |  R     | TSM writes the guest's vstimecmp to allow the host to schedule an internal software timer for the guest. 
+| vsie       |  W    |  R     | TSM writes the guest's vsie to allow the host to check which interrupts are enabled. This is useful
+                                in waking up a guest's vcpu when it's sleeping due to a `WFI` instruction.
+| *GPRs*     |       |        |
+| a0         |  RW   |  RW    | TSM transforms the load or store instruction to/from a0 before writing to the htinst CSR.
+                                So a0 is the only GPR used for MMIO emulation. Also, a0 is used to pass the ECALL argument and return the result for ECALLs forwarded to the host.
```

Comment:
you already described here. But this is a salus specific thing. So should be a recommendation.